sasl/aws_msk_iam_v2/msk_iam.go

```diff
@@ -32,15 +32,15 @@ const (
 	queryExpiryKey   = "X-Amz-Expires"
 )
 
-var signUserAgent = fmt.Sprintf("kafka-go/sasl/aws_msk_iam/%s", runtime.Version())
+var signUserAgent = fmt.Sprintf("kafka-go/sasl/aws_msk_iam_v2/%s", runtime.Version())
 
 // Mechanism implements sasl.Mechanism for the AWS_MSK_IAM mechanism, based on the official java implementation:
 // https://github.com/aws/aws-msk-iam-auth
 type Mechanism struct {
 	// The sigv4.Signer of aws-sdk-go-v2 to use when signing the request. Required.
 	Signer *signer.Signer
-	// The aws.Credentials of aws-sdk-go-v2. Required.
-	Credentials aws.Credentials
+	// The aws.CredentialsProvider of aws-sdk-go-v2. Required.
+	CredentialsProvider aws.CredentialsProvider
```

Comment:
`Credentials` is shorter and clearly communicates the purpose, and the type information already expresses that we expect a provider; how about keeping the original field name?

It's also the name used in the `aws.Config` type, so it would remain consistent.